### PR TITLE
phpspec 5.0 support

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by msvrtan** [comment]
Any change @ek9 of merging this in + new release? phpspec 5.0 was released yesterday

File: composer.json
Comment (sp-felix-carmona) [inline composer.json]: it should be `||` instead of `|`

**Review by sp-felix-carmona — CHANGES_REQUESTED** [review]

File: composer.json
Comment (samsonasik) [reply composer.json]: updated

**Comment by roukmoute** [comment]
Any news about that?

**Comment by samsonasik** [comment]
@roukmoute temporary workaround until it get merged, we can define in composer.json my repository fork:

```json
    "require-dev": {
        // ...
        "leanphp/phpspec-code-coverage": "dev-patch-1"
        // ...
    },
    "repositories" : [
        {
            "type" : "vcs",
            "url" : "https://github.com/samsonasik/phpspec-code-coverage",
            "no-api": true
        }
    ]
```

**Comment by roukmoute** [comment]
Yep thanks, this is what I've done.

**Comment by roukmoute** [comment]
@samsonasik I think you could do a temporary valid packagist waiting for it to unlock here.

**Comment by bartosz-maciaszek** [comment]
Any news on this?

**Review by UFOMelkor — APPROVED** [review]

**Comment by danmichaelo** [comment]
The PR seems valid. According to the [phpspec changelog](https://github.com/phpspec/phpspec/blob/master/CHANGES-v5.md) there shouldn't be any BC breaks apart the PHP version bump.

But @ek9 hasn't been active on github since July 2018, so I guess there's isn't much hope of having this accepted soon. That's quite a long github break btw., I really hope everything is fine with him/her!

**Comment by rlanting** [comment]
The domain ek9.co seems to have expired 2018-10, which makes me think nothing will happen here. The only other person I see in the commit history is @Seldaek, so perhaps he can shed some light on what is going on.

**Comment by Seldaek** [comment]
Sorry but I have no idea what this project even is, I merely submitted a PR to remove a link because I got a warning for it on packagist.org support.

**Comment by loevgaard** [comment]
@sp-felix-carmona or @UFOMelkor do you have write access? It seems @ek9 is rather inactive

**Comment by Dragonrun1** [comment]
So if anyone else is interested I decided to fork this and add my fork to Packagist. I updated it for ^5.0 as well so at least until this one comes back to life you are all welcome to use it.
composer require dragonrun1/phpspec-code-coverage

**Comment by aur1mas** [comment]
@Dragonrun1 if you're planning to maintain this bundle, consider tagging new version (to make it stable) & allowing to submit issues ;)

**Comment by Dragonrun1** [comment]
I don't have a lot of time to do any real package maintenance but since I do use it a lot I thought I'd try to help some other people out a little while I was at it. I'll look at doing a new tagged version in next couple days and if people have PRs I'll look at them as well as I have time. I've got a lot of other projects going right now so I wasn't looking at adding a new one ;)

**Comment by shulard** [comment]
Hello @Dragonrun1 !

It's really cool to have published a new packagist version. Since you haven't enough time to work on that lib, what do you think to move it under an organization with multiple maintainer in it ? It'll just allow to have more people behind the lib and avoid the same issue that we had on this one :smile:.

Personnaly, I can take some time to maintain this one.

**Comment by Dragonrun1** [comment]
@shulard 
Sounds like a great idea if you want to make one and add me to it that would work or I could but I'd like some input on what to call the group etc. and it might be a little while before I can make the change.

**Comment by jaylinski** [comment]
@shulard @Dragonrun1 I thought about using the existing phpspec organization for maintaining this package. See https://github.com/phpspec/phpspec/pull/1245.

If the phpspec-maintainers are willing to take this repository in, we wouldn't have to create a separate organization. Otherwise the plan with the organization seems like a good idea. :+1:

**Comment by shulard** [comment]
Hello @jaylinski !

You're absolutely right, if we can move that repository in the main phpspec organization, it'll help a lot. Let see what they think about it.

**Comment by shulard** [comment]
Hello @jaylinski !

You're absolutely right, if we can move that repository in the main phpspec organization, it'll help a lot. Let see what they think about it.

**Comment by jaylinski** [comment]
> Hello @jaylinski !
> 
> You're absolutely right, if we can move that repository in the main phpspec organization, it'll help a lot. Let see what they think about it.

Seems they are inactive too... :roll_eyes:

**Comment by BigZ** [comment]
please ! we need this one !

**Comment by LewisMcFly** [comment]
Any update? We need it as well.

**Comment by mamazu** [comment]
Well the build job is failing. So we might need to fix that first.

**Comment by samsonasik** [comment]
The travis build failure in php 7.2 env is nothing todo with the changes. It was travis issue.

**Comment by mamazu** [comment]
Ah then they need to restart the job.

**Comment by jaylinski** [comment]
> Hello @jaylinski !
> 
> You're absolutely right, if we can move that repository in the main phpspec organization, it'll help a lot. Let see what they think about it.

Turns out they don't like the idea...

**Review by Dragonrun1 — APPROVED** [review]

**Comment by Dragonrun1** [comment]
Sorry I've been busy with school etc but finally, have a little time to try to move at least the merge forward.

**Comment by Dragonrun1** [comment]
Sorry thought this was on my fork guess not but while I'm here if there are any other ideas on improving the project I might have a little time right now to look into working on some stuff so ideas are welcome.

**Comment by drupol** [comment]
Based on #38 , there is now a fork that support PHPSpec 5: https://packagist.org/packages/drupol/phpspec-code-coverage

**Comment by BigZ** [comment]
as phpspec doesn't want it and the maintainer is away from github since more than a year, i think it is time for this library to find a new home.
any one with solid maintaining capabilities willing to take this one ?

**Comment by shulard** [comment]
I’l not sure about solid maintaining capabilities but I’ll be glad to help ;-)

To avoid the same issue with the only commiter/owner who left the project, I think the best is to create an organisation like phpspec-extension. Maybe we’ll have the chance to aggregate multiple extension maintainers in it...

WDYT ?
I already created the org and forked the project in it. Anyone who want to be involved in that project can be added...

**Comment by drupol** [comment]
I'll be glad to help maintaining the fork I did.

**Comment by shulard** [comment]
@drupol thank you for the fork but what do you think about the organisation ? Because it’s always better to not rely on individuals directly for such things...

Won’t it be better to add you as a maintainer on the org repo ?

Your version is currently the more advanced one so we can start working from that version if you want ?

**Comment by shulard** [comment]
Also for the communication and share relying on the phpspec-extensions name can be clearer and help being more confident with the support...

**Comment by drupol** [comment]
> @drupol thank you for the fork but what do you think about the organisation ? Because it’s always better to not rely on individuals directly for such things...
> 
> Won’t it be better to add you as a maintainer on the org repo ?

Definitely. I'll be glad to do so. 
However, i'm abroad right now with limited access to the Internet, so, we'll discuss this further in a week :)

**Comment by bartosz-maciaszek** [comment]
`friends-of-phpspec` ;) I'm also happy to help.

**Comment by shulard** [comment]
Glad to hear this and yes the org name can be updated :-)

Same on my side limited access to the web until Friday but we’ll discuss this ASAP.

**Comment by jaylinski** [comment]
@shulard Thanks for taking the lead on this. I'm also happy to help as a org maintainer.

**Comment by shulard** [comment]
Hello guys !

Little update, the org was renamed `friends-of-phpspec` and the master branch for the fork was updated to reflect the @drupol version.

You can visit the org here: https://github.com/friends-of-phpspec

@drupol, @jaylinski you have been added as member of the org and you have access to the repository. I'll create some issues / PR to be sure that everything will be migrated.

**Comment by shulard** [comment]
Hello everyone !

I'm glad to announce that I've just published a new official release of this package from the new [friends of PHPSpec](https://github.com/friends-of-phpspec) organization !

If you want a revamped and maintained version (compatible with PHPSpec 5.0), update your composer.json and move forward 😉.

https://packagist.org/packages/friends-of-phpspec/phpspec-code-coverage

Currently, everything is compatible so a quick composer.json update will make it works smoothly...

**Comment by samsonasik** [comment]
@shulard thank you, I can close it ;)

**Comment by shulard** [comment]
@Dragonrun1, if you still don't want to maintain your own release of this package, can you mark it abandoned on packagist in favor of the new one ?

**Comment by Dragonrun1** [comment]
I'll look at updating my fork today. And have done so with a link to the new version.

**Pull request closed without merging**
